torch/_torch_docs.py

```diff
@@ -948,6 +948,101 @@ def merge_dicts(*dicts):
 Alias for :func:`torch.atanh`.
 """)
 
+add_docstr(torch.asarray,
+           r"""
+asarray(obj, *, dtype=None, device=None, copy=None, requires_grad=False) -> Tensor
+
+Converts :attr:`obj` into a tensor, sharing data and preserving autograd history
```

Comment:
could you please disable `asarray` for tensors with conj and neg bit set to `True`? see https://github.com/pytorch/pytorch/pull/61925 which disables `numpy()` for similar cases

Reply:
Hi @anjali411, thank you for the review.

Maybe I'm missing something here, but I don't think we need it here.
As far as I understand it, it should be disabled if we are: (1) transforming the tensor into something else (e.g. NumPy array); or (2) interpreting the tensor bytes as some other dtype (e.g. by calling `view(dtype)`). Is that correct?

If so, this function does neither. If the input is a tensor, we wrap the same tensor into another Python object (i.e. no view with different dtypes are created). If a different dtype is provided, it calls `tensor.to(other_dtype)` (which, I believe, does not disables those tensors).

I'm not too familiar with those bit flags, so the problem might be that I just don't understand how they work. Sorry about that.
Could you explain to me a little bit more the reason why this function needs to be disabled for those tensors?

Reply:
@ysiraichi `tensor.numpy()` creates a view, meaning it returns a numpy array object that's viewing the same memory as `tensor`. I assume this to be true for `asarray` as well. A simple way to test this would be to perform an in-place operation on the `tensor` after `asarray` is called:

```py
x = torch.randn(2)
y = x.asarray()
x.zero_()
# if it prints zero, then it validates that y views the same memory as x
print(y) 
```